### ISSUE TITLE
[BUG] Rancher webhook validations fails on EKS clusters with calico 

### ISSUE THREAD

**Issue by pancochea** [rancher/rancher#41172]
**Rancher Server Setup**
- Rancher version: 2.7.2
- Installation option (Docker install/Helm Chart): Helm Chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): EKS v1.23.16-eks-48e63af
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: v1.23.16-eks-48e63af
- Cluster Type (Local/Downstream): Any
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Imported for rancher, eks for downstream

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) Admin
  - If custom, define the set of permissions:

**Describe the bug**
There is a known caveat when using Calico over EKS that all communications between in-cluster resources and kubeapi calls must happen in the vpc network. This includes Validation and mutation webhooks. To solve this, the hostNetwork parameter must set to true for all webhooks. 

The now mandatory rancher-webhook component does not have this option active, thus any change in the cluster related with the new ValidationWebhooks fail with the following error:

`Internal error occurred: failed calling webhook "[rancher.cattle.io](http://rancher.cattle.io/).secrets": failed to call webhook: Post "[https://rancher-webhook.cattle-system.svc:443/v1/webhook/mutation/secrets?timeout=10s](https://rancher-webhook.cattle-system.svc/v1/webhook/mutation/secrets?timeout=10s)": Address is not allowed`

To fix this, you have to set the global.hostNetwork to true in the "app" installed as rancher-webhook in the rancher cluster and the downstream clusters.

There isn't any option to directly configure the webhook with that option, either in the rancher helm chart, or in the configuration for the downstream clusters.

I expect an option in the cluster configuration to set the webhook's hostNetwork to true, both in the rancher-helm chart and in the cluster configuration part for the downstream clusters.

**To Reproduce**
Install rancher in an EKS cluster with calico as CNI, then run any change affected by the Validation webhook and it will fail as kubeapi cannot contact rancher-webhook.

**Result**
`Internal error occurred: failed calling webhook "[rancher.cattle.io](http://rancher.cattle.io/).secrets": failed to call webhook: Post "[https://rancher-webhook.cattle-system.svc:443/v1/webhook/mutation/secrets?timeout=10s](https://rancher-webhook.cattle-system.svc/v1/webhook/mutation/secrets?timeout=10s)": Address is not allowed`

**Screenshots**

**Additional context**
More information about the "hostNetwork" need for calico on EKS:

Calico networking cannot currently be installed on the EKS control plane nodes. As a result the control plane nodes will not be able to initiate network connections to Calico pods. (This is a general limitation of EKS's custom networking support, not specific to Calico.) As a workaround, trusted pods that require control plane nodes to connect to them, such as those implementing admission controller webhooks, can include hostNetwork:true in their [pod spec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#podspec-v1-core). See the Kubernetes API pod spec definition for more information on this setting.

source: https://docs.tigera.io/calico/latest/getting-started/kubernetes/managed-public-cloud/eks

**Comment by pgonin** [comment]
Could you have a look at https://github.com/rancher/rancher/issues/41191
and new documentation from Tigera  https://github.com/tigera/docs/pull/678
It appears related to introduction of PSAs

**Comment by pancochea** [comment]
@pgonin I don't think that it's related. My issue is solved by activating the hotnetwork in the webhook deployment, #41191 seems to be solved by adding clusterrole and its bindings.

What this issue needs, is that the "hostNetwork" option is exposed to the upstream cluster.

**Comment by wzrdtales** [comment]
had same bug, also needed to activate hostNetwork for the webhook in the downstream cluster

**Comment by wzrdtales** [comment]
this bug was introduced with 2.7.x, with 2.6.x everything was fine

**Comment by mjura** [comment]
It seems that PSA fix https://github.com/tigera/docs/pull/678/files works fine. The question is if we want to run `rancher-webhook` with `hostNetwork` for me it is decreasing security level. Final decision belongs to Rancher Webhook developers.

**Comment by pancochea** [comment]
The cluster role has nothing to do, it's a network issue. I agree that enabling the hostNetwork by default is not the best approach, but there should be an option to enable it from the rancher cluster creation process.

**Comment by aliokaitis** [comment]
Setting these on vlaues file should work:

webhook:
  global:
    hostNetwork: true
  port: 9443
